### PR TITLE
haskell-language-server: init at 0.1.0.0

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
###### Motivation for this change
Add haskell-language-server to nixpkgs
https://github.com/haskell/haskell-language-server

###### Things done (after WIP)

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GuillaumeDesforges** [comment]
Currently, build fails

```
Building library for haskell-language-server-0.1.0.0..
[ 1 of 17] Compiling Ide.Logger       ( src/Ide/Logger.hs, dist/build/Ide/Logger.o )
[ 2 of 17] Compiling Ide.Cradle       ( src/Ide/Cradle.hs, dist/build/Ide/Cradle.o )
[ 3 of 17] Compiling Ide.Plugin.Config ( src/Ide/Plugin/Config.hs, dist/build/Ide/Plugin/Config.o )
[ 4 of 17] Compiling Ide.PluginUtils  ( src/Ide/PluginUtils.hs, dist/build/Ide/PluginUtils.o )
[ 5 of 17] Compiling Ide.Types        ( src/Ide/Types.hs, dist/build/Ide/Types.o )
[ 6 of 17] Compiling Ide.Plugin.Formatter ( src/Ide/Plugin/Formatter.hs, dist/build/Ide/Plugin/Formatter.o )

src/Ide/Plugin/Formatter.hs:67:33: error:
    • Couldn't match expected type ‘shake-0.19:Development.Shake.Internal.Core.Types.Action
                                      (FileVersion, Maybe T.Text)
                                    -> IO (a1, Maybe T.Text)’
                  with actual type ‘IO a0’
    • The first argument of ($) takes one argument,
      but its type ‘IO a0’ has none
      In a stmt of a 'do' block:
        (_, mb_contents) <- runAction "Formatter" ideState
                              $ getFileContents fp
      In the expression:
        do (_, mb_contents) <- runAction "Formatter" ideState
                                 $ getFileContents fp
           case mb_contents of
             Just contents -> do ...
             Nothing
               -> return
                    $ Left
                        $ responseError
                            $ T.pack
                                $ "Formatter plugin: could not get file contents for " ++ show uri
   |
67 |             (_, mb_contents) <- runAction "Formatter" ideState $ getFileContents fp
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Ide/Plugin/Formatter.hs:67:55: error:
    • Couldn't match expected type ‘shake-0.19:Development.Shake.Internal.Core.Types.Action
                                      a0’
                  with actual type ‘IdeState’
    • In the second argument of ‘runAction’, namely ‘ideState’
      In the expression: runAction "Formatter" ideState
      In a stmt of a 'do' block:
        (_, mb_contents) <- runAction "Formatter" ideState
                              $ getFileContents fp
   |
67 |             (_, mb_contents) <- runAction "Formatter" ideState $ getFileContents fp
   |                                                       ^^^^^^^^
[16 of 17] Compiling Paths_haskell_language_server ( dist/build/autogen/Paths_haskell_language_server.hs, dist/build/Paths_haskell_language_server.o )
[17 of 17] Compiling Ide.Version      ( src/Ide/Version.hs, dist/build/Ide/Version.o )
builder for '/nix/store/hyyg8f9srgh7xwzadlbin34jkgzfsm15-haskell-language-server-0.1.0.0.drv' failed with exit code 1
error: build of '/nix/store/hyyg8f9srgh7xwzadlbin34jkgzfsm15-haskell-language-server-0.1.0.0.drv' failed
```

**Comment by maralorn** [comment]
Hey @GuillaumeDesforges,

cool that you are working on this.

I am a bit apprehensive about deploying a software for which the developers intentionally decided not to do a release yet. Especially since we might expected additional maintenance overhead. But that‘s not on me to decide. I‘d actually love to test hls, so why not.^^

I will just give you a bit of feedback inline. I hope that‘s okay even if this is still WIP. Feel free to ignore.

**Comment by maralorn** [comment]
btw. can you please make haskell PRs against the haskell-updates branch? It will get merged into master once a week, if we are sure that everything works fine.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 1]
```diff
@@ -1506,4 +1506,55 @@ self: super: {
   # through $PATH but they aren't in $PATH
   dhall-lsp-server = dontCheck super.dhall-lsp-server;
 
+  # hasn't bumped upper bounds
+  floskell = markUnbroken (dontCheck (doJailbreak super.floskell));
```
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1510]: markUnbroken is not optimal on the haskell-updates branch.

Better solution is to remove floskell from the broken list in configuration-hackage2nix.yaml.

For tests a quick comment on why it‘s unreasonable to fix in nixpkgs would be great (although not mandatory).

For jailbreaks an issue on upstream about bumping the bounds would be awesome.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 2]
```diff
@@ -1506,4 +1506,55 @@ self: super: {
...
+
+  # hasn't bumped upper bounds
+  monad-dijkstra = markUnbroken (dontCheck (doJailbreak super.monad-dijkstra));
```
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1513]: same as above

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 3]
```diff
@@ -1506,4 +1506,55 @@ self: super: {
...
+
+  # packages with the right version, required for haskell-language-server
```
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1515]: For getting packages with the right version, you should pin those in `configuration-hackage2nix.yaml` in the extra-packages list.

Then they would automatically be added to hackage-packages.nix when hackage2nix is run.

You can run hackage2nix yourself by cloning https://github.com/nixos/cabal2nix and using the regenerate-nixpkgs.sh script. (Please tell me if the documentation in the script is unclear.)

**Comment by GuillaumeDesforges** [comment]
Thanks for the review @maralorn , it is always appreciated !

As it is a WIP PR I might not apply everything right now (since it's still changing/moving around a lot) but once it stabilizes I will properly do all the above. Let's keep things clean!

**Comment by GuillaumeDesforges** [comment]
@maralorn Sorry to bother you, but it seems like `./Setup build` from `buildPhase` makes a `dist` folder... while `./Setup test` expects a `dist-newstyle` folder. Does it ring a bell to you ?

https://gist.github.com/GuillaumeDesforges/7da3253decd1e5b3ac8776cd0f184747

**Comment by maralorn** [comment]
Can‘t say that I have encountered this one before.

I mean the general issue is the difference between cabal v1-build (which uses dist) and cabal v2-build (which uses dist-newstyle). So maybe you can fix this by using `./Setup v2-build` or something?

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 4]
```diff
@@ -1507,5 +1507,39 @@ self: super: {
...
+  lsp-test_0_11_0_2 = dontCheck super.lsp-test_0_11_0_2;
+
+  # haskell-language-server uses its own fork
+  hls-ghcide = 
+    dontCheck (
+      (self.callPackage (import ../tools/haskell/haskell-language-server/hls-ghcide.nix) { }).override {
+        # specific versions required
+        lsp-test = self.lsp-test_0_11_0_2;
+      }
+    );
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1531]: If appropriate, could you report these failed tests (and upper bound problems) upstream, and then link inline to the upstream issues?

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 5]
```diff
@@ -1507,5 +1507,39 @@ self: super: {
...
+
+  # only used by haskell-language-server as of now
+  # Tests disabled: there is an issue with Cabal 3
+  #   -> it builds in `./dist` while it needs `./dist-newstyle` for the tests
+  lsp-test_0_11_0_2 = dontCheck super.lsp-test_0_11_0_2;
+
+  # haskell-language-server uses its own fork
+  hls-ghcide = 
+    dontCheck (
+      (self.callPackage (import ../tools/haskell/haskell-language-server/hls-ghcide.nix) { }).override {
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/configuration-common.nix:1527]: I think this `callPackage` should go into `non-hackage-packages.nix` (although the `dontCheck` and `lsp-test` can probably be left here in this file).

File: .gitignore [hunk 6]
```diff
@@ -19,3 +19,6 @@ result-*
 
 # generated by pkgs/common-updater/update-script.nix
 update-git-commits.txt
+
+# VS Code workspace files
+.vscode/
```
Comment (cdepillabout) [inline .gitignore:24]: Could you remove this and send it in a different PR?

I imagine it is probably okay to add this, but I don't know anything about vscode, or if there are any rules here in nixpkgs about adding to the `.gitignore` file.

**Review by cdepillabout — CHANGES_REQUESTED** [review]
@GuillaumeDesforges Thanks for working on this!

I think this is looking pretty good.  I left a few comments on things to fix up, but in general this is looking good.

**Comment by GuillaumeDesforges** [comment]
Thanks @cdepillabout for the feedback, will do it right away :+1:

File: pkgs/development/tools/haskell/haskell-language-server/default.nix [hunk 7]
```diff
@@ -0,0 +1,67 @@
+{ mkDerivation, fetchgit
```
[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:1]: What do you mean by "update" ? 
My understanding is that thoses derivations depend on their github repositories, so updating means changing current revision. Anything else seems to be the "common" haskell nixpkgs infrastructure.

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 8]
```diff
@@ -183146,6 +183142,17 @@ self: {
...
+  "opentelemetry_0_4_0" = callPackage
+    ({ mkDerivation, base, bytestring, exceptions }:
+     mkDerivation {
+       pname = "opentelemetry";
+       version = "0.4.0";
+       sha256 = "1wp6r6625qangiw3lk2nj64khbb0jnp7rxb8s78q37icpdax9ck8";
+       libraryHaskellDepends = [ base bytestring exceptions ];
+       license = stdenv.lib.licenses.asl20;
+       hydraPlatforms = stdenv.lib.platforms.none;
+     }) {};
```
[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply pkgs/development/haskell-modules/hackage-packages.nix]: This was a leftover from an earlier version of this PR. I ran hackage2nix to fix this, sorry about that.

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 9]
```diff
@@ -1507,5 +1507,39 @@ self: super: {
...
+  haskell-language-server = 
+    super.haskell-language-server.override {
+      # specific versions required
+      ormolu = super.ormolu_0_0_5_0;
+      lsp-test = self.lsp-test_0_11_0_2;
+      # hls uses a forked ghcide
+      ghcide = self.hls-ghcide;
+      # requires the actual git to run tests, not the git haskell package
+      # this is a dirty workaround, but I can't see any other way
+      git = pkgs.git;
```
[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply pkgs/development/haskell-modules/configuration-common.nix:1542]: I'll be less negative in the comments then, thanks

[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/development/haskell-modules/configuration-common.nix:1542]: Ah, I'm sorry, I didn't mean that you need to be less negative in the comments, but just that `git = pkgs.git` isn't really a workaround.  It's just what you have to do, and it's basically the only way to do it (without adding some more logic in `callPackage` / `override`).

(If anything, I think there deserves to be _more_ negativity in the comments in nixpkgs, haha.)

File: pkgs/development/tools/haskell/haskell-language-server/default.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:1]: Maybe I'm not understanding the structure here.

So you've generated `pkgs/development/tools/haskell/haskell-language-server/default.nix` and `pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix` by hand with `cabal2nix`?

Ideally there would be an easy way to regenerate these files with `cabal2nix`, so that someone could easily update it if a new version of one of these packages was released to github.

Without having a (semi-?)automated script, a potential future contributor would just have to know that those files were generated by `cabal2nix`, and how to run `cabal2nix` again to generate them.

[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:1]: Yes, I cloned the github repos and used cabal2nix to get those, but I also had to edit it a bit (e.g. change src, add git dependency)

I believe that those packages will be updated by 
1. Changing git rev
2. Manual trial/error process as any other nixpkgs package

Though the forked `ghcide` might disappear.

HLS is quite unstable, so I'm not ready to automate it (yet).

I will document how those files where written at the top of each

[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:1]: As long as you're using cabal2nix and the Haskell architecture in nixpkgs, it should be easy to automate the creation and updates of these two derivations.

There is often no reason to edit derivations from `cabal2nix` by hand (except as overrides in configuration-common.nix).

[the remark this answers is not in the document]
Comment (GuillaumeDesforges) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:1]: cabal2nix does miss the git dependency and the preCheck hook:

https://github.com/NixOS/nixpkgs/blob/72e5c024af9af386886d1a653189b9b0629254a5/pkgs/development/tools/haskell/haskell-language-server/default.nix#L57-L60

[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/development/tools/haskell/haskell-language-server/default.nix:5]: > cabal2nix does miss the git dependency and the preCheck hook

Ah, yeah, so these two things should be added in the `configuration-common.nix` file (if possible).

File: pkgs/development/haskell-modules/configuration-common.nix [hunk 10]
```diff
@@ -1507,5 +1507,45 @@ self: super: {
...
+  # not sure if this issue comes from nixpkgs wrapping or lsp-test itself
+  lsp-test_0_11_0_2 = dontCheck super.lsp-test_0_11_0_2;
+
+  # haskell-language-server uses its own fork
+  # Test disabled: it seems to freeze (is it just that it takes a long time ?)
+  hls-ghcide = 
+    dontCheck (
+      super.hls-ghcide.override {
+        # specific versions required
+        lsp-test = self.lsp-test_0_11_0_2;
```
Comment (maralorn) [inline pkgs/development/haskell-modules/configuration-common.nix:1533]: Have you confirmed that the dependency override is needed even after disabling tests?

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 11]
```diff
@@ -3092,6 +3092,25 @@ self: {
        license = stdenv.lib.licenses.mit;
      }) {};
 
+  "ChasingBottoms_1_3_1_9" = callPackage
```
Comment (maralorn) [inline pkgs/development/haskell-modules/hackage-packages.nix:3095]: Can you try regenerating this file with the matching all-cabal-hashes version? That would make the PR much nicer to read and decrease the likelihood of rebase issues later on.

File: pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix [hunk 12]
```diff
@@ -0,0 +1,68 @@
+# A forked version of ghcide by the haskell-language-server team
+#
+# This file was generated using cabal2nix and required manual edition,
+# there is no automated way to upgrade it.
+# 1. upgrade the git revision
+# 2. trial and error to make it work
```
Comment (cdepillabout) [inline pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix:6]: Like I said above, the automated way to upgrade this should be to just re-run `cabal2nix`.

Any sort of trail and error to make it work (like adding the git dependency, etc) should be added in the `configuration-common.nix` file, not manually in these two files.

File: pkgs/development/haskell-modules/configuration-common.nix (hunk 10 above)
Comment (GuillaumeDesforges) [reply pkgs/development/haskell-modules/configuration-common.nix:1533]: Yes

File: pkgs/development/haskell-modules/hackage-packages.nix (hunk 11 above)
Comment (GuillaumeDesforges) [reply pkgs/development/haskell-modules/hackage-packages.nix:3095]: Where can I find the instructions to do that ? It's not really clear to me sorry

**Comment by GuillaumeDesforges** [comment]
TODO
- [ ] use a update.sh script to generate `pkgs/development/tools/haskell/haskell-language-server/default.nix` and `pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix`
- [ ] make override in `pkgs/development/haskell-modules/configuration-common.nix`
- [ ] regenerating `pkgs/development/haskell-modules/hackage-packages.nix` with the matching all-cabal-hashes version

**Comment by maralorn** [comment]
>     * [ ]  regenerating `pkgs/development/haskell-modules/hackage-packages.nix` with the matching all-cabal-hashes version

Let me know if you have problems with that. We introduced `./regenerate-nixpkgs.sh` to `cabal2nix` that should help a lot with this job and I would be really interested if it is helpful for other people.

**Comment by GuillaumeDesforges** [comment]
Lost a lot of time when trying to automate the update process since cabal2nix was not fetching the same revision of the ghcide fork as the one used by hls, but now this looks pretty good!

**Comment by maralorn** [comment]
@GuillaumeDesforges Thank you for putting so much work into this! 

I have compiled and tested this on a project of mine with neovim+coc.nvim. Works like a charm.

We should probably do something like https://github.com/NixOS/nixpkgs/pull/89450 for haskell-language-server, too.

I have also tried running the `./update.sh` script. It reproduced exactly the same to nix expressions. Seems to work.

@GrahamcOfBorg test haskellPackages.haskell-language-server

**Comment by maralorn** [comment]
@GrahamcOfBorg haskellPackages.haskell-language-server

**Comment by GuillaumeDesforges** [comment]
> We should probably do something like #89450 for haskell-language-server, too.

I'm open to suggestions, but I think this can be done in another PR so that this "step" is closed and I can move on in my life ;)

**Comment by maralorn** [comment]
> > We should probably do something like #89450 for haskell-language-server, too.
> 
> I'm open to suggestions, but I think this can be done in another PR so that this "step" is closed and I can move on in my life ;)

By all means. Let‘s do that in a later PR.

**Comment by maralorn** [comment]
@GrahamcOfBorg build haskellPackages.haskell-language-server

File: pkgs/development/tools/haskell/haskell-language-server/update.sh [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+hls_new_version=$(curl --silent "https://api.github.com/repos/haskell/haskell-language-server/commits/master" | jq '.sha' --raw-output)
+
+echo "Updating haskell-language-server from old version $hls_old_version to new version $hls_new_version."
+echo "Running cabal2nix and outputting to ${hls_derivation_file}..."
+
+cabal2nix --revision "$hls_new_version" "https://github.com/haskell/haskell-language-server.git" > "$hls_derivation_file"
+
+# TODO: This should ideally also automatically update the docsSearchVersion
+# from pkgs/development/haskell/configuration-nix.nix.
+
```
Comment (cdepillabout) [inline pkgs/development/tools/haskell/haskell-language-server/update.sh:55]:
```suggestion
```

I think this was a leftover from the `spago` updater file.

**Comment by cdepillabout** [comment]
@GuillaumeDesforges This looks good!  Thanks for going back and forth with us so much on this.

@maralorn Thanks for your reviews as well.

**Comment by GuillaumeDesforges** [comment]
Thank you very much for your time and comments, I learned a lot doing this PR :)

**Comment by poscat0x04** [comment]
this package doesn't build for ghc 8.10.1 btw:

```
nix-repl> :b nixpkgs.haskell.packages.ghc8101.haskell-language-server
builder for '/nix/store/csy7bipjf6hcqkvzyfa3mksz6gaphq09-data-tree-print-0.1.0.2.drv' failed with exit code 1; last 10 log lines:
    $, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:1024:20 in Cabal-3.2.0.0:Distribution.Simple.Configure
    configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:477:12 in Cabal-3.2.0.0:Distribution.Simple.Configure
    configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:625:20 in Cabal-3.2.0.0:Distribution.Simple
    confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:65:5 in Cabal-3.2.0.0:Distribution.Simple.UserHooks
    configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:180:19 in Cabal-3.2.0.0:Distribution.Simple
    defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:116:27 in Cabal-3.2.0.0:Distribution.Simple
    defaultMain, called at Setup.hs:2:8 in main:Main
  Setup: Encountered missing or private dependencies:
  base >=4.8 && <4.14
cannot build derivation '/nix/store/4dc9dn4h0pqqwm854sz56nkfksf5b2d3-brittany-0.12.1.1.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/9x573ymz2j3387gvhbdrkibhxj0cra1j-haskell-language-server-0.1.0.0.drv': 1 dependencies couldn't be built
[1 built (1 failed), 305 copied (2342.7 MiB), 181.3 MiB DL]
error: build of '/nix/store/9x573ymz2j3387gvhbdrkibhxj0cra1j-haskell-language-server-0.1.0.0.drv' failed
```

**Comment by GuillaumeDesforges** [comment]
> this package doesn't build for ghc 8.10.1

It would be needed to make overrides in all the different GHC configurations.

It is true I only tested 8.8.3 (current "main" GHC on nixpkgs)

**Comment by maralorn** [comment]
Well that concrete issue seems likely to be solvable with one simple jailbreak and is an easy upstream issue.

**Comment by GuillaumeDesforges** [comment]
@poscat0x04 If it's an upper bound issue, could you please report to upstream?

**Comment by poscat0x04** [comment]
or we can disable brittany (https://github.com/haskell/haskell-language-server/blob/77c31c37aceb822ceb5b60f11f4933419353c675/stack-8.10.1.yaml#L26-L32)

**Comment by poscat0x04** [comment]
this is the corresponding upstream issue: https://github.com/lspitzner/brittany/issues/269

**Comment by GuillaumeDesforges** [comment]
If it is bound to be fixed, we can just wait for upstream to fix.

**Comment by 414owen** [comment]
Seems broken on GHC 8.6.5:

```
nix-repl> :b (import ../../nixpkgs {}).haskell.packages.ghc865.haskell-language-server
builder for '/nix/store/rfdc1s19l2j4ad0g79kw92dkglfa6a7f-rebase-1.6.1.drv' failed with exit code 1; last 10 log lines:
    die', called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:950:20 in Cabal-2.4.0.1:Distribution.Simple.Configure
    configureFinalizedPackage, called at libraries/Cabal/Cabal/Distribution/Simple/Configure.hs:460:12 in Cabal-2.4.0.1:Distribution.Simple.Configure
    configure, called at libraries/Cabal/Cabal/Distribution/Simple.hs:596:20 in Cabal-2.4.0.1:Distribution.Simple
    confHook, called at libraries/Cabal/Cabal/Distribution/Simple/UserHooks.hs:67:5 in Cabal-2.4.0.1:Distribution.Simple.UserHooks
    configureAction, called at libraries/Cabal/Cabal/Distribution/Simple.hs:178:19 in Cabal-2.4.0.1:Distribution.Simple
    defaultMainHelper, called at libraries/Cabal/Cabal/Distribution/Simple.hs:115:27 in Cabal-2.4.0.1:Distribution.Simple
    defaultMain, called at Setup.hs:2:8 in main:Main
  Setup: Encountered missing dependencies:
  time ==1.9.*
```

**Comment by maralorn** [comment]
@414owen at the time ghc 8.8 came out no one had ever heard of the haskell-language-server project. So I am not sure if it is very reasonable to expect this to work. That being said. Maybe you wanna try with "pkgs.haskell.lib.doJailbreak"?

**Comment by 414owen** [comment]
Cool, I'll give it a shot.
Previously I was using haskell-language-server on 8.6.5 through nix with https://github.com/korayal/hls-nix/blob/master/default.nix
Pretty happy hls is landing in nixpkgs.

**Comment by GuillaumeDesforges** [comment]
I definitly have to test on the different available GHCs, for now I only developped the expression for 8.6.5. My apologies for the lack of clarity about that.
If you could work on 8.6.5 @414owen I'd be really glad!

**Comment by 414owen** [comment]
@GuillaumeDesforges sure thing. [This line](https://github.com/NixOS/nixpkgs/pull/92777) is all it took to get it to build on 8.6.5.
